pkg/controllers/scheduling/scheduling_controller.go

```diff
@@ -0,0 +1,410 @@
+		clusterSetLister:        clusterSetLister,
+		clusterSetBindingLister: clusterSetBindingLister,
+		placementLister:         placementLister,
+		placementDecisionLister: placementDecisionInformer.Lister(),
+		rateLimiter: workqueue.NewMaxOfRateLimiter(
+			workqueue.NewItemExponentialFailureRateLimiter(5*time.Millisecond, 120*time.Second),
+			&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+		),
+	}
+
+	return factory.New().
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			key, _ := cache.MetaNamespaceKeyFunc(obj)
+			return key
+		}, placementDecisionInformer.Informer()).
+		WithSync(c.sync).
+		ToController("DecisionSchedulingController", recorder)
+}
+
+func (c *decisionSchedulingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	queueKey := syncCtx.QueueKey()
+	namespace, name, err := cache.SplitMetaNamespaceKey(queueKey)
+	if err != nil {
+		// ignore placementdecision whose key is not in format: namespace/name
+		utilruntime.HandleError(err)
+		return nil
+	}
+
+	klog.V(4).Infof("Reconciling placementdecision %q", queueKey)
+	placementDecision, err := c.placementDecisionLister.PlacementDecisions(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		// no work if placementdecision is deleted
+		c.rateLimiter.Forget(queueKey)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// no work if placementdecision is deleting
+	if !placementDecision.DeletionTimestamp.IsZero() {
+		c.rateLimiter.Forget(queueKey)
+		return nil
+	}
+
+	return c.schedule(ctx, syncCtx, placementDecision)
+}
+
+// schedule makes decisions for the unscheduled decisions in the status of PlacementDecision.
+func (c *decisionSchedulingController) schedule(ctx context.Context, syncCtx factory.SyncContext, placementDecision *clusterapiv1alpha1.PlacementDecision) error {
+	selectedClusters := sets.NewString()
+	for _, decision := range placementDecision.Status.Decisions {
+		if len(decision.ClusterName) == 0 {
+			continue
+		}
+
+		selectedClusters.Insert(decision.ClusterName)
+	}
+
+	// no work if all decisions have already been scheduled
+	queueKey := fmt.Sprintf("%s/%s", placementDecision.Namespace, placementDecision.Name)
+	numOfDecisions := len(placementDecision.Status.Decisions)
+	if numOfDecisions == selectedClusters.Len() {
+		c.rateLimiter.Forget(queueKey)
+		return nil
+	}
+
+	// get the owner placement
+	placementName := placementDecision.Labels[placementLabel]
+	if len(placementName) == 0 {
+		return fmt.Errorf("no placement label on placementdecision %q", queueKey)
+	}
+	placement, err := c.placementLister.Placements(placementDecision.Namespace).Get(placementName)
+	if errors.IsNotFound(err) {
+		// the placement must be deleted, do nothing
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// get all feasible clusters for the placement
+	feasibleClusters, err := getFeasibleClusters(placement, c.clusterLister, c.clusterSetLister, c.clusterSetBindingLister)
```

Comment:
it feels like this list should be sorted by name or some such to  help get stable selection results.